src/lib/common/sol-mainloop-impl-common.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "sol-mainloop-impl-common.h"
+#include "sol-log.h"
+#include "sol-vector.h"
+#include "sol-util.h"
+
+bool mainloop_common_run_loop;
+
+bool mainloop_common_timeout_processing;
+unsigned int mainloop_common_timeout_pending_deletion;
+struct sol_ptr_vector mainloop_common_timeout_vector = SOL_PTR_VECTOR_INIT;
+
+bool mainloop_common_idler_processing;
+unsigned int mainloop_common_idler_pending_deletion;
+struct sol_ptr_vector mainloop_common_idler_vector = SOL_PTR_VECTOR_INIT;
+
+int
+sol_mainloop_impl_timeout_compare(const void *data1, const void *data2)
+{
+    const struct sol_timeout_common *a = data1, *b = data2;
+
+    return sol_util_timespec_compare(&a->expire, &b->expire);
+}
+
+void
+sol_mainloop_impl_common_shutdown(void)
```

Comment:
I believe `_common_` infix is all that's needed for these functions: they're not platform-specific functions, so there's no need to use the `_impl_` prefix.